Programs/brlapi_client.c

```diff
@@ -1662,10 +1662,29 @@ static int getControllingTty(void)
     FILE *stream;
 
     if ((stream = fopen("/proc/self/stat", "r"))) {
-      int vt;
-      int ok = fscanf(stream, "%*d %*s %*c %*d %*d %*d %d", &tty) == 1;
+      char buf[256];
+      size_t read = fscanf(buf, sizeof(*buf), sizeof(buf) / sizeof(buf[0]), stream);
```

Comment:
You need to add a trailing `\0` at the end of the read string (and thus substract 1 from the size passed to read), otherwise `sscanf` below may overflow.

Reply:
Updated.